docs/guide/in-depth/advanced-chunks.md

```diff
@@ -1,6 +1,238 @@
+- A module will be either captured by `automatic chunking` or `manual chunking` depending on your configuration, but not both. If a module is not captured by `manual chunking`, it will be still put into a chunk which is created by `automatic chunking` with respecting rules we explained in the [code splitting](./code-splitting.md) guide.
+
+## Why use `advancedChunks`?
+
+The `automatic chunking` doesn't take loading performance or cache invalidation into account. It simply groups modules based on their static imports. This can lead to suboptimal chunking, where large chunks are created that may not be performant for loading or cause cache invalidation for every deployment.
+
+## How to use `advancedChunks`?
+
+Let's take a look at the following example:
+
+```jsx
+// index.jsx
+import * as ReactDom from 'react-dom';
+import App from './App.jsx';
+
+ReactDom.createRoot(document.getElementById('root')).render(
+  <App />,
+);
+
+// App.jsx
+import * as React from 'react';
+import { Button } from 'ui-lib';
+
+export default function App() {
+  return <Button onClick={() => alert('Button clicked!')} />;
+}
+```
+
+and you get the following output:
+
+```js [output-hash0.js]
+// node_modules/react/index.js
+'React library code';
+
+// node_modules/ui-lib/index.js
+'UI library code';
+
+// node_modules/react-dom/index.js
+'ReactDOM library code';
+
+// App.js
+function App() {
+  return <Button onClick={() => alert('Button clicked!')} />;
+}
+
+// index.js
+
+ReactDom.createRoot(document.getElementById('root')).render(<App />);
+```
+
+In this example,
+
+- We used 3 libraries: `react`, `react-dom`, and `ui-lib`.
+- `output-hash0.js` is the output file generated by Rolldown.
+- `hash0` is the hash of the output file, it changes if the content of the file changes.
+
+### Reduce cache invalidation
+
+Let's talk about cache invalidation first. Cache invalidation here means that when you deploy a new version of your application, the browser will need to download the new version of the file. If the file is large, this can lead to a poor user experience.
+
+For example, you change the `app.jsx` file
+
+```jsx [app.jsx]
+function App() {
+  return <Button onClick={() => alert('Button clicked!')} />; // [!code --]
+  return <Button onClick={() => alert('Button clicked!!!')} />; // [!code ++]
+}
+```
+
+and naturally, you get a `output-hash1.js` file with the same content as `output-hash0.js`, except the `App` function is changed.
+
+Now, if you deploy this new version of your application, the browser will need to download the entire `output-hash1.js` file, even though only a small part of it has changed. This is because the hash of the file has changed, and the browser will treat it as a new file.
+
+To solve this problem, we can use `advancedChunks` to split output libraries into separate chunks, because they don't change frequently.
```

Comment:
```suggestion
To solve this problem, we can use `advancedChunks` to split output libraries into separate chunks, because they don't change frequently compared to application code.
```